core/src/main/java/org/apache/calcite/avatica/remote/AbstractService.java

```diff
@@ -89,7 +89,7 @@ ColumnMetaData finagle(ColumnMetaData column) {
       case PROTOBUF:
         return column;
       default:
```

Comment:
I am not a big fan of default + exception pattern for the same reasons mentioned in [Don't default that switch](https://www.youtube.com/watch?v=SpMIYtl_A98). All IDEs (for sure Elciplse and Intellij) are capable of detecting easily such cases and return an error or warning depending on the configuration. How about killing it completely?

Reply:
I disagree. That video is talking about a language that (objective c?) that can check that switch statements are exhaustive. I wish that Java did this, but it does not.

In Java, an error that occurs during testing is a good result. Even an error that occurs in production is not too bad - it generates a clear stack trace, which is better than giving mysterious behavior.

If people don't add a `default: throw ...`, the compiler makes them do silly stuff like 'return null;' after the switch. So it looks as if the method can return a nullable value, even though it cannot in practice. So, `default: throw ...` is the way to get the best mileage out of javac's (limited) flow analysis.

Reply:
I agree with julian. Since in Java switch default + exception is very usual in practice, and we just need to give right info in exception stack.

Reply:
I was talking exclusively for the switch on **enumerations**. I am also in favor of fail-fast behavior so that's why I find IDE (or checkstyle) errors/warnings more immediate. Other minor point is that not having a default case means also less code to review and maintain. Having said that both coding styles are valid so if both of you prefer default + exception then that's totally fine by me.

Reply:
Just my 2c: @leesf , would you please care to add input `getSerializationType()` as long as you modify the line?

Then failure message would contain something to analyze rather than obscure `Unhandled case statement`

Reply:
I was going to suggest the same thing, right now!

Reply:
Also as a commit  message you can use something like "Improve exception message in AbstractService". Avoid the [hotfix] prefix which it is not used in our codebase.